src/ifcparse/IfcLogger.h

```diff
@@ -20,93 +20,124 @@
-public:
-	typedef enum { LOG_PERF, LOG_DEBUG, LOG_NOTICE, LOG_WARNING, LOG_ERROR } Severity;
-	typedef enum { FMT_PLAIN, FMT_JSON } Format;
-private:
-
-	// To both stream variants need to exist at runtime or should this be a 
-	// template argument of Logger or controlled using preprocessor directives?
-	static std::ostream* log1;
-	static std::ostream* log2;
-	
-	static std::wostream* wlog1;
-	static std::wostream* wlog2;
-
-	static std::stringstream log_stream;
-
-	static Severity verbosity;
-	static Format format;
-	static boost::optional<IfcUtil::IfcBaseClass*> current_product;
-	static Severity max_severity;
-
-	static boost::optional<long long> first_timepoint;
-	static std::map<std::string, double> performance_statistics;
-	static std::map<std::string, double> performance_signal_start;
-
-	static bool print_perf_stats_on_element;
-
-public:
-	static void SetProduct(boost::optional<IfcUtil::IfcBaseClass*> product);
-
-	/// Determines to what stream respectively progress and errors are logged
-	static void SetOutput(std::wostream* l1, std::wostream* l2);
-	
-	/// Determines to what stream respectively progress and errors are logged
-	static void SetOutput(std::ostream* l1, std::ostream* l2);
-
-	/// Determines the types of log messages to get logged
-	static void Verbosity(Severity v);
-	static Severity Verbosity();
-	static Severity MaxSeverity();
-
-	/// Determines output format: plain text or sequence of JSON objects
-	static void OutputFormat(Format f);
-	static Format OutputFormat();
-	
-	/// Log a message to the output stream
-	static void Message(Severity type, const std::string& message, const IfcUtil::IfcBaseInterface* instance = 0);
-	static void Message(Severity type, const std::exception& message, const IfcUtil::IfcBaseInterface* instance = 0);
-	
-	static void Notice(const std::string& message, const IfcUtil::IfcBaseInterface* instance = 0) { Message(LOG_NOTICE, message, instance); }
-    static void Warning(const std::string& message, const IfcUtil::IfcBaseInterface* instance = 0) { Message(LOG_WARNING, message, instance); }
-    static void Error(const std::string& message, const IfcUtil::IfcBaseInterface* instance = 0) { Message(LOG_ERROR, message, instance); }
-	
-	static void Notice(const std::exception& exception, const IfcUtil::IfcBaseInterface* instance = 0) { Message(LOG_NOTICE, exception, instance); }
-	static void Warning(const std::exception& exception, const IfcUtil::IfcBaseInterface* instance = 0) { Message(LOG_WARNING, exception, instance); }
-	static void Error(const std::exception& exception, const IfcUtil::IfcBaseInterface* instance = 0) { Message(LOG_ERROR, exception, instance); }
-
-	static void Status(const std::string& message, bool new_line=true);
-
-	static void ProgressBar(int progress);
-	static std::string GetLog();
-	static void PrintPerformanceStats();
-	static void PrintPerformanceStatsOnElement(bool b) { print_perf_stats_on_element = b; }
+  public:
+    typedef enum {
```

Comment:
I guess this doesn't need to be a typedef. Is there a rule for that. Not sure if we use it often.